js/server/types/User.js

```diff
@@ -0,0 +1,228 @@
+const _ = require('lodash');
+const moment = require('moment');
+const { UserTicket } = require('./UserTicket');
+const config = require('../config');
+
+// currently just a single source of truth for the User data structure.
+// May be useful for migrating the user mutation methods to in the future.
+class User {
+  constructor () {
+    this.tickets = [];
+    this.claimableRewardsOnWithdrawnAssets = 0;
+    this.dispensed = 0;
+    this.forfeited = 0;
+    this.currentTotalClaimableReward = 0;
+    this.reservedReward = 0;
+    this.totalTicketsAmountSum = 0;
+    this.claimableCommissionsOnDelegatorRewards = 0;
+    this.totalRewardAtMaturity = 0;
+    this.ticketAmountAtMaturity = 0;
+    this.yieldAtMaturity = 0;
+    this.nextRewardShare = 0;
+    this.currentYieldOnTickets = 0;
+    // dates
+    this.maturityDate = undefined;
+    this.maturityDateISO = undefined;
+    this.yearsToMaturity = undefined;
+    this.currentAPYOnTickets = undefined;
+    this.maturityDateMs = undefined;
+    // timevalues
+    this.futureReward = undefined;
+    this.nextReward = undefined;
+    this.nextRewardProjectedFutureReward = undefined;
+    this.nextRewardProjectedAPYOnTickets = undefined;
+  }
+
+  updateMaturityTimeProps (userAtFinalTimestamp, currentTimestampInMs) {
+    this.maturityDate = userAtFinalTimestamp.maturityDate;
+    this.maturityDateISO = userAtFinalTimestamp.maturityDateISO;
+    const msToMaturity =
+      userAtFinalTimestamp.maturityDateMs - currentTimestampInMs;
+    this.yearsToMaturity = msToMaturity / 1000 / 60 / 60 / 24 / 365;
```

Comment:
We should really pull in `npm.im/ms` at some point and then just do `msToMaturity / ms('1 year')`

Reply:
@ajoslin Let's try not to add dependencies when we already have code that does what the dependency would do, or when writing the code would take less time than adding the dependency

Reply:
I have like the opposite methdology but happy to stick with that on this project.